### ISSUE TITLE
SingleDatePicker doesn't fire onDateChange in IE11

### ISSUE THREAD

**Issue by mccambridge** [airbnb/react-dates#1530]
**react-dates version**
e.g. react-dates@18.4.1

**Describe the bug**
In, IE11, SingleDatePicker renders and opens on focus. But when I click on a date, the calendar disappears without updating the date in the input. This works fine on other browsers.

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser IE
 - Version 11.0.9600.19236

**Is the issue reproducible in Storybook?**
https://airbnb.io/react-dates/?selectedKind=SingleDatePicker%20%28SDP%29&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

**Comment by waldell** [comment]
Any update on this? We're having the same issue and tried importing `airbnb-browser-shims` without success.  Would be much appreciated if this could be fixed.

**Comment by majapw** [comment]
This should be fixed in v19.0.3. Please let me know if that's not the case!

**Comment by PiotrekPaprocki** [comment]
I'm afraid it still doesn't work :< 
Just tested on WM with windows 7 and ie11 
- OS: Windows 7
- Browser IE 
- Version 11.0.9600.19266
- react-dates@19.0.3

**Comment by waldell** [comment]
It works using 19.0.3 in the DateRangePicker but the SingleDatePicker still does not work.

* OS: Windows 10
* Browser IE
* Version 11.523.17134.0
* react-dates@19.0.3

**Comment by mccambridge** [comment]
Confirmed still not working in IE for SDP default example on airbnb.io/react-dates

**Comment by austinwerner358** [comment]
I am seeing this issue on react-dates 19.0.3 (was also seeing it on 18.4.1).

- **OS:** Windows 10 Enterprise x64
- **Browser:** Internet Explorer 11
- **Browser Version:** 11.2007.14393.0
- react-dates@19.0.3

No javascript console errors.

On the same computer, it works on Chrome and Edge.

**Comment by EricBrovko** [comment]
Hi there!
The method `onDateChange` is not called on **IE11** when user selects the date

**Comment by Laawe** [comment]
Having the same issue my version is ^20.0.0

**Comment by S1rFrancis** [comment]
I'm also seeing this issue in v20.0.0

**Comment by DaveCole** [comment]
To shine a little light on the subject - I only began experiencing this issue after I upgraded from `react-scripts` 1.1.5 to 2.1.5 and `react`/`react-dom` 16.4.2 to 16.8.3.  Likely babel's default requirements at this point introduced the problem.  I changed the browserlist to `not ie <= 10` and added `babel-polyfill` as well as `airbnb-browser-shims` and still experience the problem.

**Comment by DaveCole** [comment]
Actually it looks like the error started occurring when I updated from react-dates 18.1.0 to 18.1.4.

**Comment by Laawe** [comment]
So no update on this?

**Comment by rvanlaarhoven** [comment]
> Actually it looks like the error started occurring when I updated from react-dates 18.1.0 to 18.1.1.

It's working fine for me on v18.3.1 & v18.4.0 and started to be an issue from v18.4.1.

**Comment by majapw** [comment]
I'll take a look today!

**Comment by DaveCole** [comment]
@rvanlaarhoven it might be possible I updated from 18.1.0 to 18.4.1 -- what's interesting is in order to roll back I also had to roll back the package-lock.json file; changing the version in package.json alone didn't make a difference.  Maybe I just don't grasp the impact of package-lock.json well enough.  Thanks @majapw!

**Comment by ljharb** [comment]
Once this fix lands in master, I’ll backport it to v18 and v19 as well.

**Comment by Laawe** [comment]
> > Actually it looks like the error started occurring when I updated from react-dates 18.1.0 to 18.1.1.
> 
> It's working fine for me on v18.3.1 & v18.4.0 and started to be an issue from v18.4.1.

I've tried this versions and there's no response in Internet Explorer 😢

**Comment by GattoLupesco** [comment]
Hi all,
the bug is the same founded for DateRangePicker in #1516 the solution is the same: in method "onFocusOut" of SingleDatePicker you have to change:
`if (this.dayPickerContainer.contains(e.relatedTarget || e.target)) return;`
into:
`if (this.dayPickerContainer.contains(e.relatedTarget || e.target) || this.dayPickerContainer.contains(e.target)) return;`

release the fix please...

**Comment by wkgjr** [comment]
Hello, any update on this?

**Comment by sveatoslavb** [comment]
@ljharb @majapw the issue is still present. Currently using react-dates ^18.2.2 but upgrading to ^20.0.0 still didn't solve the problem.

**Comment by ljharb** [comment]
@svvb it should be fixed in v20.2. if not, please file a new issue.
